### PR TITLE
Feat: Improve find command to search with the new optional parameters (matric, reflection & studio)

### LINKED ISSUE #185: Update find to work with Matric Number, Reflection and Studio

### PR DESCRIPTION
**Pull request by jayllo-c**
Fixes #185 

Currently, contacts can now have newly added parameters with respective
prefixes for matric number, reflection and studio. However, the
predicates that are used by the find command do not support the new
optional parameters yet.

This needs to change because the basis of the find command is to make
communication between the head TA and the different groups of students
taking the course more streamlined, which means it would be more
meaningful to find contacts by the new parameters added.

Let's change the find command and the related classes to support
finding with the new optional parameters, implementing this in the same 
way that the supported parameters are currently implemented.

This will allow for the functionality to better suit the intended
user, but will also maintain the current structure of the code,
which is beneficial for future developers.

The following shows the behaviour after the command `find r/r1` is keyed into the command box.
Before:
Screenshot 2024-03-25 163304

After:
Screenshot 2024-03-25 162004

### CODE REVIEW

File: src/main/java/seedu/address/model/person/PersonDetailContainsKeywordPredicate.java [hunk 1]
```diff
@@ -45,6 +48,12 @@ public boolean test(Person person) {
         } else if (PREFIX_TAG.equals(prefix)) {
             return person.getTags().stream()
                          .anyMatch(tag -> StringUtil.containsSubstringIgnoreCase(tag.tagName, keyword));
+        } else if (PREFIX_MATRIC_NUMBER.equals(prefix)) {
```
Comment (Zer0Legion) [inline src/main/java/seedu/address/model/person/PersonDetailContainsKeywordPredicate.java:51]: Good addition! It really levels up our app. Good to see the foundation you built paying off.

**Review by delishad21 — COMMENTED** [review]
Clean and simple addition to extend this functionality. Well Done!

File: src/test/java/seedu/address/model/person/PersonDetailContainsKeywordPredicateTest.java [hunk 2]
```diff
@@ -180,6 +183,84 @@ public void test_tagDoesNotContainKeyword_returnsFalse() {
                 .withEmail("alice@email.com").withAddress("Main Street").withTags("Friend").build()));
     }
 
+    @Test
+    public void test_matricContainsKeyword_returnsTrue() {
```
Comment (Pughal77) [inline src/test/java/seedu/address/model/person/PersonDetailContainsKeywordPredicateTest.java:187]: Good comprehensive testing

**Review by Pughal77 — COMMENTED** [review]
Nice OOP. Good to see that your building on top of your already implemented OOP solutions

**Pull request merged**